pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -767,8 +804,23 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 			}
 			if !fullWildcardFound {
 				match.ServerNames = chain.sniHosts
+				sort.Strings(match.ServerNames)
 			}
 		}
+
+		if len(chain.destinationCIDRs) > 0 {
+			 cidrArray := make([]*core.CidrRange, 0)
```

Comment:
should be able to use `match.PrefixRanges` everywhere here. by default, `match.PrefixRanges` is `nil`. `append` will create a slice if necessary